container/container.go

```diff
@@ -132,41 +131,51 @@ func (container *Container) FromDisk() error {
 		return err
 	}
 
-	if err := label.ReserveLabel(container.ProcessLabel); err != nil {
-		return err
-	}
 	return container.readHostConfig()
 }
 
-// ToDisk saves the container configuration on disk.
-func (container *Container) ToDisk() error {
+// toDisk saves the container configuration on disk and returns a deep copy.
+func (container *Container) toDisk() (*Container, error) {
+	var (
+		buf      bytes.Buffer
+		deepCopy Container
+	)
 	pth, err := container.ConfigPath()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	// Save container settings
 	jsonSource, err := ioutils.NewAtomicFileWriter(pth, 0644)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer jsonSource.Close()
 
-	enc := json.NewEncoder(jsonSource)
+	w := io.MultiWriter(&buf, jsonSource)
+	if err := json.NewEncoder(w).Encode(container); err != nil {
```

Comment:
any good reason not to? I only did it this way because it seemed cleaner, but it was totally a matter of preference.